package/src/utils/add-virtual-import.ts

```diff
@@ -0,0 +1,71 @@
+import type { HookParameters } from "astro";
+import { type Plugin } from "vite";
+
+function resolveVirtualModuleId<T extends string>(id: T): `\0${T}` {
+	return `\0${id}`;
+}
+
+function createVirtualModule(name: string, content: string): Plugin {
+	const pluginName = `vite-plugin-${name}`;
+	const virtualModuleName = `${name}`;
+
+	return {
+		name: pluginName,
+		resolveId(id) {
+			if (id === virtualModuleName) {
+				return resolveVirtualModuleId(id);
+			}
+		},
+		load(id) {
+			if (id === resolveVirtualModuleId(virtualModuleName)) {
+				return content;
+			}
+		},
+	} satisfies Plugin;
+}
+
+/**
+ * This function creates a virtual import that you can use elsewhere in your integration.
```

Comment:
I think the wording is a bit misleading, it can make people think you can actually call the import inside `integration/index.ts`

Reply:
That is true, what about "Creates a Vite virtual module and updates the Astro config.".

Removes the why and just says what it does

Reply:
Sounds perfect 👍